extensions/wikia/ARecoveryEngine/ARecoveryModule.class.php

```diff
@@ -4,30 +4,48 @@ class ARecoveryModule {
 	/**
 	 * Checks whether PageFair recovery is enabled (on current wiki)
 	 *
-	 * $wgAdDriverEnablePageFairRecovery === false; // disabled on wiki
-	 * $wgAdDriverEnablePageFairRecovery === true; // enabled on wiki
-	 *
 	 * @return bool
 	 */
-	public function isPageFairRecoveryEnabled() {
+	public static function isPageFairRecoveryEnabled() {
 		global $wgUser, $wgAdDriverEnablePageFairRecovery;
 
-		return $wgUser->isAnon() && F::app()->checkSkin( [ 'oasis' ] ) && $wgAdDriverEnablePageFairRecovery === true;
+		return static::isRecoverablePage() && $wgAdDriverEnablePageFairRecovery === true;
 	}
 
 	/**
 	 * Checks whether SourcePoint recovery is enabled (on current wiki)
 	 *
-	 * $wgAdDriverEnableSourcePointRecovery === false; // disabled on wiki
-	 * $wgAdDriverEnableSourcePointRecovery === true; // enabled on wiki
+	 * @return bool
+	 */
+	public static function isSourcePointRecoveryEnabled() {
+		global $wgUser, $wgAdDriverEnableSourcePointRecovery;
+
+		return static::isRecoverablePage() && $wgAdDriverEnableSourcePointRecovery;
+	}
+
+	/**
+	 * Checks whether SourcePoint MMS is enabled (on current wiki)
+	 *
+	 * @return bool
+	 */
+	public static function isSourcePointMessagingEnabled() {
+		global $wgUser, $wgAdDriverEnableSourcePointMMS, $wgAdDriverEnableSourcePointRecovery;
+
+		return static::isRecoverablePage() && $wgAdDriverEnableSourcePointMMS && !$wgAdDriverEnableSourcePointRecovery;
+	}
+
+	/**
+	 * Checks whether should load SourcePoint bootstrap
 	 *
 	 * @return bool
 	 */
-	public function isSourcePointRecoveryEnabled() {
-		global $wgUser, $wgAdDriverEnableSourcePointRecovery, $wgAdDriverEnableSourcePointMMS;
+	public static function shouldLoadSourcePointBootstrap() {
+		return self::isSourcePointRecoveryEnabled() || self::isSourcePointMessagingEnabled();
+	}
+
+	public static function isRecoverablePage() {
+		global $wgUser;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)